forc-plugins/forc-doc/src/lib.rs

```diff
@@ -174,15 +175,23 @@ fn build_docs(
     let Command {
         document_private_items,
         no_deps,
+        experimental,
         ..
-    } = *build_instructions;
+    } = build_instructions;
     let ProgramInfo {
         ty_program,
         engines,
         manifest,
         pkg_manifest,
     } = program_info;
 
+    let experimental = ExperimentalFeatures::new(
+        &pkg_manifest.project.experimental,
+        &experimental.experimental,
+        &experimental.no_experimental,
+    )
+    .map_err(|err| anyhow::anyhow!("{err}"))?;
```

Comment:
ooks like weird design; cant exactly pinpoint what/why

Reply:
To calculate the final set of active experimental features, we need all three pieces of information. CLI input overrides the one from the package manifest file, and the CLI can specify both `--experimental` and `--no-experimental`. I don't quite see an obvious much better design that this that would provide all the three without introducing new types for grouping. The CLI input is already grouped within `CliFields` but that structure is meant to be used by `Clap`. Making `ExperimentalFeatures::new` depending on `CliFields` don't look ideal to me.

Do you see any better approach? I am open for suggestions here. 

@xunilrj As the original author, perhaps you can share your thoughts?

Reply:
I can't foresee a better design either,   just wanted to highlight the point. Can ignore the comment in this case, should not be a blocker.